src/python/pants/option/parser.py

```diff
@@ -315,84 +326,101 @@ def add_flag_val(v: Optional[Union[int, float, bool, str]]) -> None:
+                    # option name. This covers misspellings.
+                    unscoped_option_edit_distance: int = Levenshtein.distance(
                         flag_normalized_unscoped_name, other_normalized_unscoped_name
                     )
-                    if unscoped_option_levenshtein_distance <= levenshtein_max_distance:
-                        # NB: We order the matched flags by Levenshtein distance compared to the entire option string!
-                        fully_scoped_levenshtein_distance = Levenshtein.distance(
+                    if unscoped_option_edit_distance <= max_edit_distance:
+                        # NB: We order the matched flags by Levenshtein distance compared to the
+                        # entire option string.
+                        fully_scoped_edit_distance: int = Levenshtein.distance(
                             flag_normalized_scoped_name, other_normalized_scoped_name
                         )
-                        levenshtein_matching_option_names[fully_scoped_levenshtein_distance].append(
+                        edit_distance_to_matching_flags[fully_scoped_edit_distance].append(
                             other_complete_flag_name
                         )
 
             # If any option name matched or started with the invalid flag in any scope, put that
-            # first. Then, display the option names matching in order of overall edit distance, in a deterministic way.
-            all_matching_scoped_option_names = substring_matching_option_names + [
-                flag
-                for distance in sorted(levenshtein_matching_option_names.keys())
-                for flag in sorted(levenshtein_matching_option_names[distance])
+            # first. Then, display the option names matching in order of overall edit distance.
+            # Flags with the same distance will be sorted alphabetically.
+            all_matching_scoped_flags = [
+                *substring_matching_flags,
+                *itertools.chain.from_iterable(
+                    matched_flags
+                    for _, matched_flags in sorted(edit_distance_to_matching_flags.items())
+                ),
             ]
-            if all_matching_scoped_option_names:
-                matching_flags[flag_name] = all_matching_scoped_option_names
-
-        if matching_flags:
-            suggestions_message = " Suggestions:\n{}".format(
-                "\n".join(
-                    "{}: [{}]".format(flag_name, ", ".join(matches))
-                    for flag_name, matches in matching_flags.items()
-                )
+            if all_matching_scoped_flags:
+                matching_flags[flag] = all_matching_scoped_flags
+
+        scope = self._scope_str()
+        help_instructions_scope = f" {self.scope}" if self.scope != GLOBAL_SCOPE else ""
+        help_instructions = (
+            f"Run `./pants help-advanced{help_instructions_scope}` for all available options."
```

Comment:
This should be using the global option `--pants-bin-name`, but I couldn't figure out how to get that value here in `parser.py`. I don't think we can be certain that that option has been parsed yet.